src/main/java/org/openstreetmap/atlas/checks/utility/ElevationUtilities.java

```diff
@@ -314,25 +376,32 @@ public void putMap(final Location location, final short[][] map)
      */
     private synchronized short[][] loadMap(final int lat, final int lon)
     {
+        if (this.fileFetcher == null)
+        {
+            logger.error("Cannot load maps -- fileFetcher is not initialized or is null");
+            return EMPTY_MAP;
+        }
         final String filename = this.getSrtmFileName(lat, lon);
-        Path path = Paths.get(this.srtmPath, filename);
-        if (!path.toFile().isFile())
+        Optional<Resource> path = this.fileFetcher
+                .apply(Paths.get(this.srtmPath, filename).toString());
+        if (path.isEmpty())
         {
-            for (final String ext : new String[] { "zip", "gz", "xz", "bz", "bz2", "tar" })
+            for (final String ext : POSSIBLE_COMPRESSED_EXTS)
             {
-                path = Paths.get(this.srtmPath, filename + "." + ext);
-                if (path.toFile().isFile())
+                path = this.fileFetcher
+                        .apply(Paths.get(this.srtmPath, filename + "." + ext).toString());
+                if (path.isPresent())
                 {
                     break;
                 }
             }
         }
-        if (!path.toFile().isFile())
+        if (path.isEmpty())
```

Comment:
OK. This is confusing me.

What I think you are saying:
```java
        if (path.isEmpty())
        {
            for (final String ext : POSSIBLE_COMPRESSED_EXTS)
            {
                path = this.fileFetcher
                        .apply(Paths.get(this.srtmPath, filename + "." + ext).toString());
                if (path.isPresent())
                {
                    break;
                }
            }
            return EMPTY_MAP;
        }
```
That would not work, since the whole point of the for loop is to _find_ a file. We don't know if a file is found until after the for loop finishes/is broken, which means that we need to check if the path is empty again.

The alternative would be
```java
        if (path.isEmpty())
        {
            for (final String ext : POSSIBLE_COMPRESSED_EXTS)
            {
                path = this.fileFetcher
                        .apply(Paths.get(this.srtmPath, filename + "." + ext).toString());
                if (path.isPresent())
                {
                    break;
                }
            }
            if (path.isEmpty())
            {
                return EMPTY_MAP;
            }
        }
```
Which, in my opinion, is less clear. I could extract the for loop to "getPath" or something similar, at which point your suggestion might work.
```java
       private getPath() {
        Optional<Resource> path = this.fileFetcher
                .apply(Paths.get(this.srtmPath, filename).toString());
        if (!path.isEmpty()) return path;
        
            for (final String ext : POSSIBLE_COMPRESSED_EXTS)
            {
                path = this.fileFetcher
                        .apply(Paths.get(this.srtmPath, filename + "." + ext).toString());
                if (path.isPresent())
                {
                    return path;
                }
            }
        return Optional.empty();
    }

private synchronized short[][] loadMap(..., ...) {
...
    Optional<Resource> path = getPath();
    if (path.isEmpty()) return EMPTY_MAP;
....
}
```

Reply:
You're right -- I was just trying to avoid having the same `if` check twice in a row, but it's perfectly valid the way you have it. Just a styling preference on my end as it got me scratching my head for a second. Another way would be to stream POSSIBLE_COMPRESSED_EXTS, something like this after the isEmpty check on line 399---
`Arrays.stream(POSSIBLE_COMPRESSED_EXTS).map(ext -> this.fileFetcher.apply(Paths.get(this.srtmPath, filename + "." + ext).toString())).filter(Optional::isPresent).findFirst()` then return that Optional or Optional.empty()